cfpb/consumerfinance.gov
# Allow gulp watch to rebuild JS/CSS in apps

For faster development in unprocessed/apps, the gulp watch tasks are extended to see changes in apps and re-build them. Because webpack is slow, we only rebuild the modified app.

---

## Changes

- otherBuildTriggerFiles is extended to see changes in apps LESS files.
- scripts:watch and styles:watch are extended to handle apps.
- The docs now mention the gulp watch task.

## How to test this PR

1. Run `./setup.sh && yarn run gulp watch` for the initial build and starting the watch.
2. Make a change to a .less or .js file inside unprocessed/apps.
3. Check that the associated files in static_built are updated.


## Notes and todos

- Ideally I could clean up the `scriptsAppsFilter` variable after webpack finishes, but I don't see a way.

## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance
